frame/staking/src/lib.rs

```diff
@@ -2355,25 +2370,46 @@ impl<T: Config> Module<T> {
 	/// Actually make a payment to a staker. This uses the currency's reward function
 	/// to pay the right payee for the given staker account.
 	fn make_payout(stash: &T::AccountId, amount: BalanceOf<T>) -> Option<PositiveImbalanceOf<T>> {
-		let dest = Self::payee(stash);
-		match dest {
-			RewardDestination::Controller => Self::bonded(stash)
-				.and_then(|controller|
-					Some(T::Currency::deposit_creating(&controller, amount))
-				),
-			RewardDestination::Stash =>
-				T::Currency::deposit_into_existing(stash, amount).ok(),
-			RewardDestination::Staked => Self::bonded(stash)
-				.and_then(|c| Self::ledger(&c).map(|l| (c, l)))
-				.and_then(|(controller, mut l)| {
-					l.active += amount;
-					l.total += amount;
-					let r = T::Currency::deposit_into_existing(stash, amount).ok();
-					Self::update_ledger(&controller, &l);
-					r
-				}),
-			RewardDestination::Account(dest_account) => {
-				Some(T::Currency::deposit_creating(&dest_account, amount))
+		let policy = Self::payee(stash);
+		let payout = |dest: RewardDestination<T::AccountId>, amount: BalanceOf<T>| 
+			-> Option<PositiveImbalanceOf<T>> {
+			match dest {
+				RewardDestination::Controller => Self::bonded(stash)
+					.and_then(|controller|
+						Some(T::Currency::deposit_creating(&controller, amount))
+					),
+				RewardDestination::Stash =>
+					T::Currency::deposit_into_existing(stash, amount).ok(),
+				RewardDestination::Staked => Self::bonded(stash)
+					.and_then(|c| Self::ledger(&c).map(|l| (c, l)))
+					.and_then(|(controller, mut l)| {
+						l.active += amount;
+						l.total += amount;
+						let r = T::Currency::deposit_into_existing(stash, amount).ok();
+						Self::update_ledger(&controller, &l);
+						r
+					}),
+				RewardDestination::Account(dest_account) => {
+					Some(T::Currency::deposit_creating(&dest_account, amount))
+				}
+			}
+		};
+		match policy {
+			RewardPolicy::One(destination) => payout(destination,amount),
+			RewardPolicy::Split(dest1,pct,dest2) => {
+				if pct.is_one() {
+					return payout(dest1,amount)
+				}
+				if pct.is_zero() {
+					return payout(dest2,amount)
+				}
+				let first_amt = pct * amount;
+				let remaining = amount - first_amt;
```

Comment:
```suggestion
				let remaining = amount.saturating_sub(first_amt);
```
This is not needed, but I tend to prefer being safe anyway no ?